File: monitor.cpp
#include <thread>
#include <chrono>
#include <iostream>

using std::cout, std::flush;
using std::this_thread::sleep_for;
using std::chrono::seconds;

int areVitalsOk(float temp, float pulse, float spo2) {
    return checkTemperature(temp) && checkPulseRate(pulse) && checkOxygenLevel(spo2);
}

int checkTemperature(float temp) {
    if (temp < 95 || temp > 102) {
        cout << "Temperature is critical!\n";
        blinkWarning();
        return 0;
    }
    return 1;
}

int checkPulseRate(float pulse) {
    if (pulse < 60 || pulse > 100) {
        cout << "  Pulse Rate is out of range!\n";
        blinkWarning();
        return 0;
    }
    return 1;
}

int checkOxygenLevel(float spo2) {
    if (spo2 < 90) {
        cout << " Oxygen Saturation is out of range!\n";
        blinkWarning();
        return 0;
    }
    return 1;
}

void blinkWarning() {
    for (int i = 0; i < 6; i++) {
        cout << "\r* " << flush;
        sleep_for(seconds(1));
        cout << "\r *" << flush;
        sleep_for(seconds(1));
    }
}

File: monitor.py
import sys
from time import sleep

def flash_alert():
    for _ in range(6):
        print('\r* ', end='')
        sys.stdout.flush()
        sleep(1)
        print('\r *', end='')
        sys.stdout.flush()
        sleep(1)

def are_vitals_normal(temp, pulse, oxygen):
    if temp > 102 or temp < 95:
        print('Temperature is critical!')
        flash_alert()
        return False
    elif pulse < 60 or pulse > 100:
        print('Pulse rate is out of normal range!')
        flash_alert()
        return False
    elif oxygen < 90:
        print('Oxygen level is too low!')
        flash_alert()
        return False
    return True
